arcgis-ios-sdk-samples/Geometry/Spatial operations/SpatialOperationsViewController.swift

```diff
@@ -15,146 +15,159 @@
-            controller.presentationController?.delegate = self
-            controller.preferredContentSize = CGSize(width: 200, height: 200)
+    private func performOperation(_ operation: SpatialOperation) {
+        let resultGeometry: AGSGeometry?
+        switch operation {
+        case .none:
+            resultGeometry = nil
+        case .union:
+            resultGeometry = AGSGeometryEngine.union(ofGeometry1: polygon1, geometry2: polygon2)!
+        case .difference:
+            resultGeometry = AGSGeometryEngine.difference(ofGeometry1: polygon1, geometry2: polygon2)!
+        case .symmetricDifference:
+            resultGeometry = AGSGeometryEngine.symmetricDifference(ofGeometry1: polygon1, geometry2: polygon2)!
+        case .intersection:
+            resultGeometry = AGSGeometryEngine.intersection(ofGeometry1: polygon1, geometry2: polygon2)!
         }
+        // Update the geometry.
+        resultGraphic.geometry = resultGeometry
     }
     
-    // MARK: - OperationsListVCDelegate
-    
-    func operationsListViewController(_ operationsListViewController: OperationsListViewController, didSelectOperation index: Int) {
-        if index == 0 {
-            if let resultGraphic = resultGraphic {
-                self.graphicsOverlay.graphics.remove(resultGraphic)
-                self.resultGraphic = nil
+    @IBAction func chooseOperationBarButtonTapped(_ sender: UIBarButtonItem) {
+        let selectedIndex = SpatialOperation.allCases.firstIndex(where: { $0 == selectedOperation })
+        
+        let controller = OptionsTableViewController(labels: SpatialOperation.allCases.map { $0.label }, selectedIndex: selectedIndex) { [weak self] newIndex in
+            guard let self = self else { return }
+            let newOperation = SpatialOperation.allCases[newIndex]
+            if self.selectedOperation != newOperation {
+                self.selectedOperation = newOperation
+                // Perform the spatial operation.
+                self.performOperation(self.selectedOperation)
             }
-        } else {
-            self.performOperation(index)
         }
+        
+        // Configure the options controller as a popover.
+        controller.modalPresentationStyle = .popover
+        controller.presentationController?.delegate = self
+        controller.preferredContentSize = CGSize(width: 300, height: CGFloat(SpatialOperation.allCases.count) * 44)
+        controller.popoverPresentationController?.barButtonItem = sender
+        
+        // Show the popover.
+        present(controller, animated: true)
     }
     
-    // MARK: - UIAdaptivePresentationControllerDelegate
+    // MARK: UIViewController
     
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["SpatialOperationsViewController"]
```

Comment:
Since this sample now uses `OptionsTableViewController`, it should be included in the source code.